### PR TITLE
RDM-12898 - Allow requesting document not yet been added to a case

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-12898

### Change description ###

Update get casedocs endpoints such that, if, when a request to get a document is made (and the document is not added to a case record yet), the following are true:

 a valid IdAM token is provided on behalf of the invoking user

-  the document meta data contains:
   -    A jurisdiction
   -    A case type
   -    No case ID
   -    A non-NULL TTL in the future  

 Then access to the document should be granted (with no check against CCD).

Following endpoints are in the scope for this change - 

GET /cases/documents/{documentId}
GET /cases/documents/{documentId}/binary

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/ccd/documentam/controller/CaseDocumentAmControllerIT.java [hunk 1]
```diff
@@ -751,6 +873,22 @@ private Document buildDocument(final Date time, final String caseTypeId) {
             .build();
     }
 
+    private Document buildDocument(final Date time) {
```
Comment (smathangi) [inline src/integrationTest/java/uk/gov/hmcts/reform/ccd/documentam/controller/CaseDocumentAmControllerIT.java:876]: can we name explicitly like  buildDocumentWithOutCaseId(..)

File: src/main/java/uk/gov/hmcts/reform/ccd/documentam/controller/endpoints/CaseDocumentAmController.java [hunk 2]
```diff
@@ -109,15 +111,22 @@ public CaseDocumentAmController(final DocumentManagementService documentManageme
...
+                    document.getCaseId(),
+                    documentId,
+                    Permission.READ,
+                    USER_PERMISSION_ERROR,
+                    documentId.toString());
+
+            return ResponseEntity.ok(document);
+        }
+
+        return ttlIsFutureDate(document.getTtl()) ? ResponseEntity.ok(document) :  ResponseEntity.badRequest().build();
```
[the remark this answers is not in the document]
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/ccd/documentam/controller/endpoints/CaseDocumentAmController.java:125]: Of course, this detail was missing from JIRA ticket but I should have asked for clarification

👍

[the remark this answers is not in the document]
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/ccd/documentam/controller/endpoints/CaseDocumentAmController.java]: Fixed in b84a992

File: src/integrationTest/java/uk/gov/hmcts/reform/ccd/documentam/controller/CaseDocumentAmControllerIT.java (hunk 1 above)
Comment (sullivangareth) [reply src/integrationTest/java/uk/gov/hmcts/reform/ccd/documentam/controller/CaseDocumentAmControllerIT.java]: Fixed in b84a992

**Pull request merged**